R/extremes-data.R

```diff
@@ -210,7 +212,27 @@ createDataRows <-
         {
           relatedValue <- relatedSet$value
         }
+        
+        if (!isEmptyOrBlank(x$relatedPrimary)) {
+          if (nrow(x$relatedPrimary) > nrow(x$points) ||
+              nrow(x$points) > nrow(x$relatedPrimary)) {
```

Comment:
is that a thing?! i'm actually writing a whole separate function now to reduce the duplicity of these two sets of statements as well and will try that..